New modal to add custom WMS basemap

viddo opened the pull request:
Implement the WMS basemap category:
![wms-basemaps](https://cloud.githubusercontent.com/assets/978461/7754269/acdfae70-ffeb-11e4-92f5-49aa6eeef144.gif)

![screen shot 2015-05-21 at 19 13 20](https://cloud.githubusercontent.com/assets/978461/7754519/8a8dbaf4-ffed-11e4-89df-b05c5bae2e89.png)

Disable layers that are already added, indicated in list-header text too (per sketch file):
![screen shot 2015-05-22 at 11 56 09](https://cloud.githubusercontent.com/assets/978461/7768641/89176554-007d-11e5-882e-697dd099ca84.png)

Some doubts and questions:
- [x] ~~How/where should error feedback be displayed?~~
  - ~~Layer failed to be saved for whatever reason (required prerequisite step before basemap is changed)~~
  - Added the standard error view
- [x] ~~In the old modal there's a back button, shouldn't there be one for this one too? If so, where?~~
  - Added back button
- [x] ~~Search is not a must-have feature, so postpone it to solve it separately, this amount of changes are already rather big for this one~~
  - To be solved in separate issue #3738
- [x] ~~As seen in screenshot about, the error box has limited space so linked to [docs page](http://docs.cartodb.com/cartodb-editor.html#including-an-external-basemap) regarding the [limited projections](https://groups.google.com/forum/#!msg/cartodb/yf81f1b_2m4/acnOnlozJBUJ) (needs to be updated), do you have another proposal on how to solve it?~~ 
  - Created issue on docs repo, will solve later since it doesn't have to be solved right away --> https://github.com/CartoDB/docs/issues/243
- [x] ~~I started looking into the WMTS-issues as reported in #3694, but it turned out to not be so trivial to fix, I suggest we postpone/solve this separately.. (remove the WTMS labels in the templates for now)~~
  - postponed, solved separately

Cartofante commented:
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3819/))

viddo commented:
@saleiva see list of doubts/questions in the description, what do you think?

Cartofante commented:
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3820/))

saleiva commented:
1 - Standard error that we have. The one that is fullscreen.
2 - Yep, in the same place we have in the privacy window for example.
3 - Cannot we reuse what pablo did for the users?
4 - I like what you did.
5 - ok.

Cartofante commented:
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3831/))

viddo commented:
@saleiva Thx for the replies, fixed the issues and updated description w/ changes. 

Regarding point 3/search, yeah, possibly we can but I'm not sure if it's implemented with the idea of being completely generic. I need to check the solution w/ @alonsogarciapablo and possibly extract it from the privacy view in that case. Regardless, reusing existing code or not, it's something that can be done in isolation why I proposed to postpone it.

viddo commented:
@xavijam can you do the code review now that the changes are done?

xavijam commented on lib/assets/javascripts/cartodb/models/user.js:
```diff
@@ -6,6 +6,10 @@ cdb.admin.UserLayers = cdb.admin.Layers.extend({
+  custom: function() {
+    return this.where({ category: undefined });
   }
 });
```
Not sure if we should move user-layers model to other file, in the case of it grows more.

viddo replied:
Agree, ideally one file one concept. But considering some things are sensitive to order it's loaded (since using the old concat of the source files) I didn't want to change that right now.

xavijam replied:
ok!

xavijam commented:
Good to go!
